From 2ee57f9b0d8d08c554392e6b832a335593a7555d Mon Sep 17 00:00:00 2001
From: Max Kellermann <max.kellermann@gmail.com>
Date: Mon, 4 Jul 2022 19:19:31 +0200
Subject: [PATCH] output/PipeWire: add nullptr check, fixing crash with
 PipeWire 0.3.53

Since PipeWire 0.3.53, control names can apparently be nulled, leading
to crashes in applications assertion that the string cannot be
nullptr.

Closes https://github.com/MusicPlayerDaemon/MPD/issues/1558
---
 NEWS                                        | 2 ++
 src/output/plugins/PipeWireOutputPlugin.cxx | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/output/plugins/PipeWireOutputPlugin.cxx b/src/output/plugins/PipeWireOutputPlugin.cxx
index 18821b1059..c752820d01 100644
--- a/src/output/plugins/PipeWireOutputPlugin.cxx
+++ b/src/output/plugins/PipeWireOutputPlugin.cxx
@@ -235,7 +235,8 @@ class PipeWireOutput final : AudioOutput {
 				[[maybe_unused]] uint32_t id,
 				const struct pw_stream_control *control) noexcept {
 		auto &o = *(PipeWireOutput *)data;
-		if (StringIsEqual(control->name, "Channel Volumes"))
+		if (control->name != nullptr &&
+		    StringIsEqual(control->name, "Channel Volumes"))
 			o.ControlInfo(control);
 	}
 
